core/sql/executor/ExHbaseIUD.cpp

```diff
@@ -1285,7 +1285,6 @@ ExWorkProcRetcode ExHbaseAccessBulkLoadPrepSQTcb::work()
       matches_ = 0;
       currRowNum_ = 0;
       numRetries_ = 0;
-      hFileParamsInitialized_ = FALSE;
```

Comment:
I'm not sure what you mean by adding another condition. Sampling is controlled by a syntax option of bulk load.

Resetting the flag for ALL_DONE is not a problem because it is only executed when the load (preparation phase) is done.  On the other hand, removing the one in NOT_STARTED, which is executed after each flush, improved load time for a wide, 52M-row table from 3.5 hours to 35-40 minutes (without Suresh's fix that changed flush frequency).

I'm not sure why Khaled thought a better mechanism was needed. I would be fine with removing the comment.